worlds/kh1/__init__.py

```diff
@@ -0,0 +1,289 @@
+            elif attempts >= 200:
+                item_names = []
+                attempts = 0
+            else:
+                attempts = attempts + 1
+
+        self.multiworld.itempool += item_pool
+
+    def pre_fill(self) -> None:
+        goal_dict = {
+            "sephiroth":       "Olympus Coliseum Defeat Sephiroth Ansem's Report 12",
+            "unknown":         "Hollow Bastion Defeat Unknown Ansem's Report 13",
+            "postcards":       "Traverse Town Mail Postcard 10 Event",
+            "final_ansem":     "Final Ansem",
+            "puppies":         "Traverse Town Piano Room Return 99 Puppies Reward 2",
+            "final_rest":      "End of the World Final Rest Chest"
+        }
+        self.multiworld.get_location(goal_dict[self.options.goal.current_key], self.player).place_locked_item(self.create_item("Victory"))
+        if self.options.vanilla_emblem_pieces:
+            self.multiworld.get_location("Hollow Bastion Entrance Hall Emblem Piece (Flame)", self.player).place_locked_item(self.create_item("Emblem Piece (Flame)"))
+            self.multiworld.get_location("Hollow Bastion Entrance Hall Emblem Piece (Statue)", self.player).place_locked_item(self.create_item("Emblem Piece (Statue)"))
+            self.multiworld.get_location("Hollow Bastion Entrance Hall Emblem Piece (Fountain)", self.player).place_locked_item(self.create_item("Emblem Piece (Fountain)"))
+            self.multiworld.get_location("Hollow Bastion Entrance Hall Emblem Piece (Chest)", self.player).place_locked_item(self.create_item("Emblem Piece (Chest)"))
+
+    def get_filler_item_name(self) -> str:
+        weights = [data.weight for data in self.fillers.values()]
+        return self.random.choices([filler for filler in self.fillers.keys()], weights)[0]
+
+    def fill_slot_data(self) -> dict:
+        slot_data = {"xpmult": int(self.options.exp_multiplier)/16,
+                    "required_reports_eotw": self.determine_reports_required_to_open_end_of_the_world(),
+                    "required_reports_door": self.determine_reports_required_to_open_final_rest_door(),
+                    "door": self.options.final_rest_door.current_key,
+                    "seed": self.multiworld.seed_name,
+                    "advanced_logic": bool(self.options.advanced_logic),
+                    "hundred_acre_wood": bool(self.options.hundred_acre_wood),
+                    "atlantica": bool(self.options.atlantica),
+                    "goal": str(self.options.goal.current_key)}
+        if self.options.randomize_keyblade_stats:
+            min_str_bonus = min(self.options.keyblade_min_str.value, self.options.keyblade_max_str.value)
+            max_str_bonus = max(self.options.keyblade_min_str.value, self.options.keyblade_max_str.value)
+            min_mp_bonus = min(self.options.keyblade_min_mp.value, self.options.keyblade_max_mp.value)
+            max_mp_bonus = max(self.options.keyblade_min_mp.value, self.options.keyblade_max_mp.value)
```

Comment:
It is probably a good idea to actually modify the option values as well here, so that the spoiler log can match what's actually being used.
```suggestion
            min_str_bonus = min(self.options.keyblade_min_str.value, self.options.keyblade_max_str.value)
            max_str_bonus = max(self.options.keyblade_min_str.value, self.options.keyblade_max_str.value)
            self.options.keyblade_min_str.value = min_str_bonus
            self.options.keyblade_max_str.value = max_str_bonus
            min_mp_bonus = min(self.options.keyblade_min_mp.value, self.options.keyblade_max_mp.value)
            max_mp_bonus = max(self.options.keyblade_min_mp.value, self.options.keyblade_max_mp.value)
            self.options.keyblade_min_mp.value = min_mp_bonus
            self.options.keyblade_max_mp.value = max_mp_bonus
```